webApp/src/main/java/com/infoshare/controller/VolunteerController.java

```diff
@@ -28,9 +32,23 @@ public String searchForVolunteer(Model model) {
         return getTestViewWithPageUnderConstructionMessage(model);
     }
 
+    @PostMapping("/submit-edited-form")
+    public String submitEditedVolunteerForm(@Valid @ModelAttribute("volunteerForm") VolunteerForm volunteerForm, BindingResult br, Model model) {
+        if (br.hasErrors()) {
+            model.addAttribute("types", volunteerService.getTypesOfHelp());
+            return "edited-volunteer-form";
+        } else {
+            volunteerService.editVolunteerData(volunteerForm.getName(), volunteerForm.getLocation(), volunteerForm.getEmail(), volunteerForm.getPhone(), volunteerForm.getTypeOfHelp(), volunteerForm.getIsAvailable(), volunteerForm.getUuid());
```

Comment:
Jest jakiś problem w logice związanej z zapisem danych po edycji, bo zamiast zmienić w pliku dane wolontariusza, to pojawia się nowy wpis, zawsze na pierwszej pozycji (na załączonym screenie są dwa wpisy 'Mariusz' - na ostatnim miejscu ten przed edycją, a na pierwszym miejscu po edycji):
![image](https://user-images.githubusercontent.com/25279453/105406064-37d5b180-5c2c-11eb-8e6a-48b589c961b3.png)

Reply:
Myślę że to dlatego, że ten plik testowy do file DB, nie jest 'dobry'. Są tam takie same UUID, przez co nie jest możliwy poprawny zapis, bo zapis który zrobiłam działa na podstawie UUID (żeby można był zmienić mail). Miałam ten sam problem, musiałam wygenerować nowe dane przez aplikacje konsolową.

Reply:
zgodnie z radą @ZytaWisz poprawiłam fileDB, tak żeby UUID się dobrze generowało, teraz powinno być OK